Lib/xmlrpc/server.py

```diff
@@ -386,30 +386,31 @@ def _dispatch(self, method, params):
         not be called.
         """
 
-        func = None
         try:
             # check to see if a matching function has been registered
             func = self.funcs[method]
         except KeyError:
-            if self.instance is not None:
+            try:
                 # check for a _dispatch method
-                if hasattr(self.instance, '_dispatch'):
-                    return self.instance._dispatch(method, params)
-                else:
-                    # call instance method directly
-                    try:
-                        func = resolve_dotted_attribute(
-                            self.instance,
-                            method,
-                            self.allow_dotted_names
-                            )
-                    except AttributeError:
-                        pass
+                func = self.instance._dispatch
+            except AttributeError:
+                # call instance method directly
+                try:
+                    func = resolve_dotted_attribute(
+                        self.instance,
+                        method,
+                        self.allow_dotted_names
+                    )
+                except AttributeError:
+                    raise Exception('method "%s" is not supported' % method)
+            else:
+                # _dispatch method found, but it accepts params as a single arg; let's change that
+                dispatch_func = func
 
-        if func is not None:
-            return func(*params)
-        else:
-            raise Exception('method "%s" is not supported' % method)
+                def func(*params):
+                    return dispatch_func(method=method, params=params)
```

Comment:
> It is not required that the dispatch function supports keyword arguments.
Agreed, changed.

> What is the purpose of this change? Why not call the dispatch function directly?
The purpose of this change is to make sure no exception handling is in progress when the function is called, thus I do attempt to first resolve the function (handling exceptions as I go) and then once the function is (hopefully) resolved, I attempt to call it

I did add `@functools.wraps` around the wrapper (forgot to check that in earlier)